NetSuite 11.0 New (#1335)
mulesoft/docs-connectors: netsuite/11.0/modules/ROOT/pages/netsuite-connector-upgrade-migrate.adoc

= Upgrade to NetSuite Connector 11.x
:page-aliases: connectors::netsuite/netsuite-connector-upgrade-migrate.adoc

Upgrade NetSuite Connector to version 11.x to use the new features.

== Supported Upgrade Paths

[%header,cols="50a,50a"]
|===
|From Version | To Version
|10.x |11.x
|===

== Changes in This Release

> In the v10.x versions of NetSuite Connector, the connector resolved metadata by receiving Java objects as input and converting them internally to XML so that NetSuite could understand them. In the 11.x version, NetSuite Connector resolves XML metadata directly.

Reviewer: ```suggestion
In the NetSuite Connector 10.x, the connector resolved metadata by receiving Java objects as input and converting them internally to XML so that NetSuite could understand them. In NetSuite Connector 11.x, XML metadata is resolved directly.
```

> Because of this modification, the input and output metadata for connector operations has changed and is now equivalent to the request and response bodies of the corresponding SOAP API operation, respectively. You can use DataSense explorer in Studio to drag and drop the metadata fields, which causes their corresponding namespaces and tags to be generated automatically.

Reviewer: ```suggestion
Because of this modification, the input and output metadata for connector operations has changed and is now equivalent to the request and response bodies of the corresponding SOAP API operations. You can use DataSense explorer in Studio to drag and drop the metadata fields, which generates their corresponding namespaces and tags automatically.
```

Version 11.x also eliminates the REST-based operations.

[[changed-sources]]
== Changes to Sources

> Removed the Page Size field for the sources (On Deleted Object, On Modified Object, and On New Object). Each source now returns one record in the output metadata.

Reviewer: ```suggestion
Removed the **Page Size** field for the sources On Deleted Object, On Modified Object, and On New Object. Each source now returns one record in the output metadata.
```

[[changed_operations]]
== Changes to Operations

=== Advanced Operations

Changes to advanced operations include the following:

[%header%autowidth.spread]
|===
| Operation |Changes
| Change Email | Returns a NetsuiteSoapAttributes object in the output metadata
| Change Password | Returns a NetsuiteSoapAttributes object in the output metadata
| Get Account Governance Info | Returns a GetAccountGovernanceInfoResult object in the output metadata
| Get Data Center URLs | Returns a NetsuiteSoapAttributes object in the output metadata.
| Get Integration Governance Info | Returns a GetIntegrationGovernanceInfoResult object in the output metadata
| Get Server Time | Returns a NetsuiteSoapAttributes object in the output metadata
|===

=== Async Operations

> Each operation now return a NetsuiteSoapAttributes object in the output metadata. Other changes to the async operations include the following:

Reviewer: ```suggestion
Each operation now returns a `NetsuiteSoapAttributes` object in the output metadata. Other changes to the async operations include the following:
```

| Async Get List | Removed the Retrieved Base Ref Types field. You now specify the associated values in the Message field of the input metadata.
| Async Get List Records | Removed this operation. For equivalent functionality, use the Get List operation and specify `FILE` as the record type in the Message field of the input metadata.
| Async Initialize List | Removed the Initialize Records field. You now specify the associated values in the Message field of the input metadata.
| Async Search | Removed the Search Record and Criteria fields. You now specify the associated values in the Message field of the input metadata.
| Async Update List | Renamed Record Type to Type. Removed Records attributes. You now specify the associated values in the Message field of the input metadata.
| Async Upsert List | Renamed Record Type to Type. Removed Records attributes. You now specify the associated values in the Message field of the input metadata.
| Check Async Status | No changes.
| Get Async Result | No changes.
|===

=== Invitee operations

> Each invitee operation now returns a NetsuiteSoapAttributes object in the output metadata. Other changes to the invitee operations include the following:

Reviewer: Should it be Invitee operations?

* Returns a NetsuiteSoapAttributes object in the output metadata.
| Update Records List | Removed this operation. For equivalent functionality, use the Update List operation.
|Upsert List a| * Removed the Record Maps field. You now specify the associated values in the Message field of the input metatada.
* Returns a NetsuiteSoapAttributes object in the output metadata.
| Upsert Record a a| * Renamed the operation to Upsert and removed the Attributes field. You now specify the associated values in the Message field of the input metatada.
* Returns a NetsuiteSoapAttributes object in the output metadata.
|===

== Upgrade Steps

If you are upgrading from an earlier version of the connector, you can click the `Updates Available` pop-up Message in the bottom right corner of Anypoint Studio and follow the prompts.

You can also follow these steps to perform the upgrade:

. In Anypoint Studio, create a Mule project.
. In *Mule Palette*, click *Search in Exchange*.
. In *Add Dependencies to Project*, enter `netsuite` in the search field.
. In *Available modules*, select *NetSuite* and click *Add*. +
Studio upgrades the connector automatically.
. Click *Finish*.

> . In the `pom.xml` file for the Mule project, verify that the NetSuite Connector dependency version is `10.0.0`.

Reviewer: ```suggestion
. In the `pom.xml` file for the Mule project, verify that the NetSuite Connector dependency version is `11.0.0`.
```
